### ISSUE TITLE
[Feature]: Allow multiple logits

### ISSUE THREAD

**Issue by blainenelson** [cisco-foundation-ai/faith#26]
### Problem Statement

Currently, we only assess the first logit in LOGITS mode.  It would be ideal to allow for processing multiple logits to go beyond MCQA.

### Proposed Solution

Adapt grading code to consider solutions using multiple logits.

### Checklist

- [x] I have read the [contributing guidelines](/RobustIntelligence/faith/blob/main/CONTRIBUTING.md)
- [x] I have verified this does not duplicate an existing feature request
